api/google-search/src/main/java/com/google/search/googlesearch/controller/v1/api/SearchAutoCompleteController.java

```diff
@@ -0,0 +1,35 @@
+package com.google.search.googlesearch.controller.v1.api;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.google.search.googlesearch.models.SearchAutoCompleteModel;
+import com.google.search.googlesearch.services.SearchAutoCompleteService;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+
+@RestController
+@RequestMapping(value = "/api/v1/")
+public class SearchAutoCompleteController {
+	
+	@Autowired
+	private SearchAutoCompleteModel searchAutoCompleteModel;
+	
+	@Autowired
+	private SearchAutoCompleteService searchAutoCompleteService;
+	
+	@GetMapping(value = "search/autocomplete",produces = "application/json")
+	@ApiResponses(value = {
+			@ApiResponse(responseCode = "200", description = "Autocomplete Search keyword (goo,ama,fac,gma, you)"),
+			@ApiResponse(responseCode = "400", description = "Invalid Request"),
+			@ApiResponse(responseCode = "500", description = "Internal server error")})
+	public ResponseEntity<SearchAutoCompleteModel> autoCompleteSearch(@RequestParam(defaultValue = "goo") String q) {
+		searchAutoCompleteModel.getResults().addAll(searchAutoCompleteService.findBySearchKeyword(q));
+		return ResponseEntity.ok(searchAutoCompleteModel);
```

Comment:
Use proper HTTP code for expected and unexpected results.

Reply:
Added `HTTP.FOUND` and `HTTP.NOT_FOUND `for the results. Need your review on it.

Reply:
@ramsunvtech : Spring Boot itself handles 400 and 500 responses. On a different note maybe we shall consider coding an custom `Exception handler` to have control over the error message. 

@jpukg : Let us know your thought on it.

Reply:
@jpukg This is something important for us to understand. every API. lets say.

Search Keywords - POST
200 - EMPTY RESPONSE / RESPONSE.
400 - BAD INPUT - which means empty keyword in this case.
500 - INTERNAL_SERVER_ERROR - when DB / down stream is not responding accessible.

Registration -later we can discuss this one
Login - later we can discuss this one